DB/pgsql.php

```diff
@@ -1110,6 +1127,33 @@ function _checkManip($query)
                 || parent::_checkManip($query));
     }
 
+    // }}}
+    // {{{ _resultId()
+
+    /**
+     * Returns a numeric value identifying the result, used for maintaining
+     * indexes. Required by migration from resources to 'PgSql\Result'
+     * types with PHP 8.1.
+     *
+     * @param mixed $result  the query result resource or PgSql\Result
+     *
+     * @return int
+     *
+     * @access protected
+     */
+    function _resultId($result)
+    {
+        return
+            is_resource($result)
+                ? ( function_exists('get_resource_id')
+                    ? get_resource_id($result)
+                    : (int)$result
+                ) : ( function_exists('spl_object_id')
+                    ? spl_object_id($result)
+                    // catch-all statement
+                    : (int)$result
+                )
```

Comment:
@DanCld After a quick check on PHP5, there is a parsing error, because of missing semicolon in line 1155.
Could you add that, please?